notebooks/data-sources/TestGrid/metrics/metric_template.ipynb

```diff
@@ -0,0 +1,1052 @@
+{
```

Comment:
<p>This groupby would group tests from different tabs and grids into the same group (assuming there are tests whose names don't vary across tabs/grids), and then calculate the % flakes. Is that what we're going for here, or would <code>df.groupby(['tab', 'grid', 'test'])</code> be a more accurate measurement?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aicoe-aiops/ocp-ci-analysis/pull/169/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

Reply:
<p>@<a href="https://github.com/chauhankaranraj" target="_blank">chauhankaranraj</a> I  think here we do want to make a claim about how "flaky" a specific test is across all grids. But good point about noting the difference, perhaps there would be some value in highlight the difference in flake severity for a test across all grids (baseline??) vs a specific grid.    </p>

Reply:
<p>I think df.groupby(['tab', 'grid', 'test']) would give flake % for a test over time which is useful and will be calculated in the Superset dashboard. The flake severity metric (Issue https://github.com/aicoe-aiops/ocp-ci-analysis/issues/138)  aims to look at a particular test across tabs and grids to see if it's flaky everywhere or not. Let me know if that seems unreasonable, we could redo the logic and update the issue if required. </p>

Reply:
> I think here we do want to make a claim about how "flaky" a specific test is across all grids

Gotcha, just wanted to make sure there's no sneaky groupby errors :)

> But good point about noting the difference, perhaps there would be some value in highlight the difference in flake severity for a test across all grids (baseline??) vs a specific grid.

Agreed, it seems worth looking into

Reply:
> I think df.groupby(['tab', 'grid', 'test']) would give flake % for a test over time which is useful and will be calculated in the Superset dashboard. The flake severity metric (Issue https://github.com/[/issues/138](https://github.com/aicoe-aiops/ocp-ci-analysis/issues/138)) aims to look at a particular test across tabs and grids to see if it's flaky everywhere or not. Let me know if that seems unreasonable, we could redo the logic and update the issue if required.

Yeah that makes sense :+1: